text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,623 @@
+  invalidate Rust safety properties, as evaluated by the Rust compiler team.
+  For example, if Rust relies on a specific code generation feature to support
+  a memory safety property, the code generation for the target should support
+  that feature. If this requirement does not hold, the target must clearly and
+  prominently document any such limitations, such as via a failing test in the
+  testsuite and/or a footnote in the target tier list, and the Rust compiler
+  team must be satisfied with the balance between these limitations and the
+  difficulty of implementing the necessary features.
+- If the target supports C code, the new Rust target should support the C
+  calling convention for the platform via `extern "C"`. The C calling
+  convention does not need to be the default Rust calling convention for the
+  target, however.
+- The target must build reliably in CI.
+- The approving teams may additionally require that a subset of tests pass in
+  CI, such as enough to build a functional "hello world" program, `./x.py test
+  --no-run`, or equivalent "smoke tests". In particular, this requirement may
+  apply if the target builds host tools, or if the tests in question provide
+  substantial value via early detection of critical problems.
+- Building the target in CI must not take substantially longer than the current
+  slowest target in CI. This requirement is subjective, to be evaluated by the
+  infrastructure team, and will take the community importance of the target
+  into account.
+- New tier 2 targets should, if at all possible, support cross-compiling. New
+  tier 2 targets should not require using the target as the host for builds,
+  even if the target supports host tools.
+- In addition to the legal requirements for all targets (specified in the tier
+  3 requirements), because a tier 2 target typically involves the Rust project
+  building and supplying various compiled binaries, incorporating the target
+  and redistributing any resulting compiled binaries (e.g. built libraries,
+  host tools if any) must not impose any onerous license requirements on any
+  members of the Rust project, including infrastructure team members and those
+  operating CI systems. This is a subjective requirement, to be evaluated by
+  the approving teams.
+  - As an exception to this, if the target's primary purpose is to build
+    components for a Free and Open Source Software (FOSS) project licensed
+    under "copyleft" terms (terms which require licensing other code under
+    compatible FOSS terms), such as kernel modules or plugins, then the
+    standard libraries for the target may potentially be subject to copyleft
+    terms, as long as such terms are satisfied by Rust's existing practices of
+    providing of full corresponding source code. Note that anything added to
+    the Rust repository itself must still use Rust's standard license terms.
+- Tier 2 targets must not impose burden on the authors of pull requests, or
```

Comment:
I assume for tier 2 (and maybe tier 3) a relevant Rust team member may decide that breaking that target is not acceptable for some reason and that it is required for the PR author to address the breakage, right? In other words maintainers _of the target_ are not allowed to block merging a PR due to breaking a tier 2 target but relevant Rust team members are.

Reply:
A Rust team may do so, but doing so implies something about the relative importance of the target, and we should aim for tiers to reflect that importance.